cJSON.c

```diff
@@ -452,20 +431,22 @@ static cJSON_bool print_number(const cJSON * const item, printbuffer * const out
     {
         length = sprintf((char*)number_buffer, "null");
     }
-    else if ((fabs(floor(d) - d) <= DBL_EPSILON) && (fabs(d) < 1.0e60))
```

Comment:
I don't think it works that way because a lot of doubles are bigger than `LIMIT_FOR_TEST_PRINTING` but should still be recoverable with just 15 decimal places of precision.

Anyway, if the printing of Integers in engineering notation is only a special case for numbers bigger than 10^15 and smaller than -10^15 (or somewhere around that) I can live with it. I though this happened to more numbers than just that.

In this case let's go for simplicity and keep the code as it is. Just remove the special case for negative zero and it can be merged.